### ISSUE TITLE
Incorrect geometry for IfcTrimmedCurve with big radian trims.

### ISSUE THREAD

**Issue by Artoymyp** [xBimTeam/XbimEssentials#92]
Hello!
Some of our partners are generating Ifc-files with IfcTrimmedCurves that have parametric trims with big radian values (> 2 pi). As far as I know there is no formal Ifc rule forbidding this.

Xbim interprets these values as degrees, and generates wrong geometry.
The root of problem is in function IfcStore.CalculateModelFactors(). Specifically in the code block under comment 
//check if angle units are incorrectly defined, this happens in some old models

My question is: what are these "old models"? 
Are they numerous? How old are they? What is their generating application? 
In other words, I am not sure should I be bothered breaking support for them in my application?

Maybe we should create a new branch in function IfcStore.SetWorkArounds() for these old files?

**Comment by martin1cerny** [comment]
What are units of the model? If `PLANEANGLEUNIT` is defined to be degrees than parameter should be degrees. If it is radians than it should be in radians. Can you check the units assignment?

**Comment by Artoymyp** [comment]
Thank you for the rapid answer.
This is correct: project's PLANEANGLEUNIT is RADIAN. 
PS: All trims' parameter values are in range [-1; 10]. 
Some of them explicitly have values of 1,5709..., 3.1415..., 6,2831...  etc.
This model is correctly displayed using Solibri and Bim Vision.

**Comment by martin1cerny** [comment]
Can you share the model please? I'll point @SteveLockley to this issue. It would be better to ask in XbimGeometry project actually.

**Comment by Artoymyp** [comment]
I have made a stripped model (the most I can provide). It perfectly demonstrates the issue:
[TrimmedCircleWall.ifc.pdf](https://github.com/xBimTeam/XbimEssentials/files/1094789/TrimmedCircleWall.ifc.pdf)
PS: This file is NOT a pdf-file... it is just a renamed ifc-file. GitHub forum does not accepts my ifc or zip files for some reason.

Actual wall:
![image](https://user-images.githubusercontent.com/12625469/27435113-dd87383c-577c-11e7-94af-3fc587023b91.png)

Expected wall (made using fix https://github.com/Artoymyp/XbimEssentials/commit/f554610720aa748c7e6342a606636a4acedba102):
![image](https://user-images.githubusercontent.com/12625469/27435092-cd95f544-577c-11e7-8151-2c3515265cf3.png)

**Comment by martin1cerny** [comment]
Can you also confirm which version of xBIM Essentials and xBIM Geometry do you use?

**Comment by Artoymyp** [comment]
I have reproduced the issue using fresh XbimWindowsUI-master. So I can say that I am using Xbim.Essentials.4.0.7 and Xbim.Geometry.4.0.11.

**Comment by SteveLockley** [comment]
This is now fixed in the currently release version 5
